detections/endpoint/multiple_users_failing_to_authenticate_using_explicit_credentials.yml

```diff
@@ -0,0 +1,46 @@
+name: Multiple Users Failing To Authenticate Using Explicit Credentials
+id: e61918fa-9ca4-11eb-836c-acde48001122
+version: 1
+date: '2021-04-13'
+author: Mauricio Velazco, Splunk
+type: batch
+datamodel:
+- Endpoint
+description: 'The following analytic identifies a source user failing to authenticate with multiple users using explicit credentials on a host.
+  This behavior could represent an adversary performing a Password Spraying attack against an Active Directory environment to obtain initial access or elevate privileges. 
+  In this logic, the number of unique users is hardcoded to `10` but can/should be customized to fit especific environments.
+  Event 4648 is generated when a process attempts an account logon by explicitly specifying that account`s credentials. This event generates on domain controllers, member servers, and workstations.
+
+  This detection will trigger on the potenfially malicious host, perhaps controlled via a trojan or operated by an insider threat, from where a
+  password spraying attack is being executed.
+
+  The analytic`s returned fields allow analysts to investigate the event further by providing fields like source account, attempted user accounts and the endpoint were the behavior was identified.'
+
+search: ' `wineventlog_security` EventCode=4648 | eval Source_Account = mvindex(Security_ID, 0) | eval Destination_Account = mvindex(Account_Name, 1) | stats  dc(Destination_Account) AS unique_accounts values(Account_Name) as tried_account by ComputerName | where unique_accounts  > 10 | `multiple_users_failing_to_authenticate_using_explicit_credentials_filter`'
```

Comment:
* Where is a failure component of user auths here - as it is now,  this detection is not looking at failures at all? Event 4648 is mostly a "success" or "info" event. Consequently, please remove this detection.

* Again, Source Account is not Security ID, it would be "mvindex(Account_Name, 0)".

* We have to do pre-filtering because this event is not only about src/dest users: In fact, in src/dest account fields we'd also have devices and service/system accounts. All those should be removed from consideration, i.e., we should add filter to look only the cases where mvindex(Account_Name, 0) and mvindex(Account_Name, 1) do not contain dollar "$" at the end, AND are not equal to SYSTEM. See this the Confluence page for examples: https://confluence.splunk.com/pages/viewpage.action?spaceKey=PROD&title=Active+Directory%3A+Identities+and+Locations+of+Logging+for+the+Entities+Reported+in+AD+events

Reply:
I updated the query based on the feedback. Also changed the name. This event does not confirm if users are failing to authenticate, just the attempt.